### PR TITLE
feat: session summaries

### PR DESCRIPTION
**Pull request by pauldambra**
A quick spike of summarizing sessions to seek feedback from everyone

it's relatively slow and costs money so I've wrapped it in a flag and am only summarizing one recording at a time

### CODE REVIEW

**Comment by pauldambra** [comment]
@neilkakkar shall we smash this into prod so I can see how it works on real recordings?

File: frontend/src/lib/api.ts [hunk 1]
```diff
@@ -1537,6 +1537,12 @@ const api = {
             return await new ApiRequest().recording(recordingId).withAction('persist').create()
         },
 
+        async summarize(
+            recordingId: SessionRecordingType['id']
+        ): Promise<{ content: string; ai_result: Record<string, any> }> {
+            return await new ApiRequest().recording(recordingId).withAction('summarize').create()
```
Comment (marandaneto) [inline frontend/src/lib/api.ts:1543]: What's about data privacy? Is everything being sent to OpenAI?
I believe this should be disabled by default with a huge note when enabling it.

Comment (pauldambra) [reply frontend/src/lib/api.ts:1543]: Absolutely... This is behind a flag and not production ready

Right now we only send event names, urls, and a text from som HTML elements.

Comment (marandaneto) [reply frontend/src/lib/api.ts:1543]: console logs and network requests (no headers or payload) would be super useful as well.

**Comment by marandaneto** [comment]
@pauldambra 

> costs money

I believe the best approach here is a project config where the customer sets their own OpenAPI Key, so this goes from their quota, it's not scalable using ours if we actually wanna beta test this, I guess.

**Comment by marandaneto** [comment]
Besides the flag, it'd be cool if we could enable/disable it via project config as well, that can be another PR though.
I like how this is going, and probably time-saving if you don't wanna watch a 30-minute recording, with a glance, you have an idea if it's worth watching it or not.

**Review by marandaneto — APPROVED** [review]
Since its behind a flag, I'd like to see how this behaves in production, LGTM

File: frontend/src/lib/api.ts (hunk 1 above)
Comment (marandaneto) [reply frontend/src/lib/api.ts:1543]: > text from som HTML elements

That's what I am afraid of, even if we use only within our org, if the text form contains an email/name, it is already a problem, we'd need something like [this](https://docs.sentry.io/product/issues/issue-details/ai-suggested-solution/#data-processing).

**Pull request merged**